File: frontend/angular_ecommerce/src/app/common/order-item.ts
import { CartItem } from "./cart-item";

export class OrderItem {
    imageUrl:string;
    unitPrice:number;
    quantity:number;
    productId:string;

//constructor pour populate le order-item based on CartItem (il faut juste les assigner)
    constructor(cartItem: CartItem){
        this.imageUrl=cartItem.imageUrl;
        this.quantity=cartItem.quantity;
        this.unitPrice=cartItem.unitPrice;
        this.productId=cartItem.id;
    }
}

File: backend/src/main/java/com/siteweb/ecommerce/entity/OrderItem.java
package com.siteweb.ecommerce.entity;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;


//Entité pour récupérer les commandes dans a database 

@Entity
@Table(name="order_item")
@Getter
@Setter
public class OrderItem {
	
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(name="id")
	private Long id;
	
	@Column(name="image_url")
	private String imageUrl;
	
	@Column(name="unit_price")
	private BigDecimal unitPrice;
	
	@Column(name="quantity")
	private int quantity;
	
	@Column(name="product_id")
	private Long productId;
	
	//Reference back de order (plusieur item dans une commande)
	@ManyToOne
	@JoinColumn(name="order_id")
	private Order order;
	
	

}
